core/chains/evm/types/models.go

```diff
@@ -167,12 +167,16 @@ func (h *Head) ChainHashes() []common.Hash {
 	return hashes
 }
 
-func (h *Head) LatestFinalizedHead() commontypes.Head[common.Hash] {
```

Comment:
I don't think I like this change. IMHO this is not consistent with what we have now. If you take `GetParent` for example we don't return an error. The notion of returning nil is quite different than returning an error. Error indicates something went wrong, while returning nil means that something wasn't initialized **yet**. Now on the Head Tracker we check if the call returned an error, when we could just check for nil and make the `LatestFinalizedHead` simpler by returning nil and not an error.